packages/domain/lib/use_case/get_movies_use_case.dart

```diff
@@ -26,36 +38,48 @@ class GetMoviesUseCase
 
     final paginationCheck =
         int.parse(response.headers[UrlConstantsDomain.pagination][0]);
-    if (paginationCheck >= 5) {
-      int itemCount = 50;
-
-      if (type == MovieType.trending) {
-        final responseWithItem = await _repository.getDataTrending(
-          itemCount: itemCount,
-        );
-        addToList(responseWithItem);
-      } else if (type == MovieType.anticipated) {
-        final responseWithItem = await _repository.getDataAnticipated(
-          itemCount: itemCount,
-        );
-        addToList(responseWithItem);
-      }
-    } else {
-      final itemCount = paginationCheck;
-
-      if (type == MovieType.trending) {
-        final responseWithItem = await _repository.getDataTrending(
-          itemCount: itemCount,
-        );
-        addToList(responseWithItem);
-      } else if (type == MovieType.anticipated) {
-        final responseWithItem = await _repository.getDataAnticipated(
-          itemCount: itemCount,
-        );
-        addToList(responseWithItem);
+
+    final String dateResponse = response.headers['date'][0];
+    final checkData = await _preferences.saveDate(dateResponse);
+    if (checkData == true){
+      if (paginationCheck >= 5) {
+        int itemCount = 50;
+
+        if (type == MovieType.trending) {
+          final responseWithItem = await _traktRepository.getDataTrending(
+            itemCount: itemCount,
+          );
+          addToList(responseWithItem);
```

Comment:
no need in addToList method just make a method with return responseWithItem for example:

final List<MovieResponse> jsonMovies = await fetchMoviesFromApi(MovieType type, int itemsCount);

Reply:
additionally,  separate logic in this class, by methods please